lib/bandit/headers.ex

```diff
@@ -46,22 +46,23 @@ defmodule Bandit.Headers do
     end
   end
 
-  @spec get_content_length(Plug.Conn.headers()) :: {:ok, nil | integer()} | {:error, String.t()}
+  @spec get_content_length(Plug.Conn.headers()) ::
+          {:ok, nil | non_neg_integer()} | {:error, String.t()}
   def get_content_length(headers) do
     case get_header(headers, "content-length") do
       nil -> {:ok, nil}
       value -> parse_content_length(value)
     end
   end
 
-  @spec parse_content_length(binary()) :: {:ok, length :: integer()} | {:error, String.t()}
+  @spec parse_content_length(binary()) :: {:ok, non_neg_integer()} | {:error, String.t()}
   defp parse_content_length(value) do
     case parse_integer(value) do
-      {length, ""} when length >= 0 ->
+      {length, ""} ->
         {:ok, length}
 
-      {length, rest} ->
-        if rest |> Plug.Conn.Utils.list() |> all?(to_string(length)),
+      {length, _rest} ->
+        if value |> Plug.Conn.Utils.list() |> all?(to_string(length)),
```

Comment:
Previously this had a bug where it would still work for an invalid content-length header like `123 123`